### PR TITLE
123669 handle invalid dates

### PR DESCRIPTION
**Pull request by eliasyishak**
Refers to issue: https://github.com/flutter/flutter/issues/123669

My detailed explanation is in this comment: https://github.com/flutter/flutter/issues/123669#issuecomment-1488908243

## Pre-launch Checklist

- [x] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes 
- [x] I have added/updated relevant documentation 
- [x] I have run "optimize/organize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could

### CODE REVIEW

**Comment by eliasyishak** [comment]
@GroovinChip let me know if I did this correctly for the PR. I was not sure if I should be merging into dev or customer_testing.

Also for the fix, it was even simpler to just set the `_selectedDay` variable to be 1 so it always points to the beginning of the month. Let me know if this is an okay assumption to make.

**Comment by GroovinChip** [comment]
@eliasyishak Thanks for submitting this fix. You're correct to target the `dev` branch. I'll review this now.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## [1.11.2]
+* Default the `_selectedDay` state variable to be 1 when selecting the previous/next month from widget to ensure new date is valid for `_formatAsDateTime()` method
```
Comment (GroovinChip) [inline CHANGELOG.md:2]:
```suggestion
* Default the `_selectedDay` state variable to be 1 when selecting the previous/next month from widget to ensure new date is valid for `_formatAsDateTime()` method (https://github.com/flutter/flutter/issues/123669 & https://github.com/macosui/macos_ui/pull/401)
```

**Comment by eliasyishak** [comment]
> @eliasyishak Thanks for submitting this fix. You're correct to target the `dev` branch. I'll review this now.

Oh but it seems like I targeted `customer_testing`, should I point it to `dev` instead?

**Comment by GroovinChip** [comment]
> > @eliasyishak Thanks for submitting this fix. You're correct to target the `dev` branch. I'll review this now.
> 
> Oh but it seems like I targeted `customer_testing`, should I point it to `dev` instead?

Ah yes. Please target dev.

**Review by GroovinChip — CHANGES_REQUESTED** [review]
Looks fine, please just update the version in `pubspec.yaml` to match your changelog entry.

**Comment by GroovinChip** [comment]
@eliasyishak Which Flutter channel are you on? see https://github.com/macosui/macos_ui/blob/dev/CONTRIBUTING.md#branch-structure

**Comment by eliasyishak** [comment]
Is it better for me to make the change on a fork of `dev` instead? I was getting a lot of errors when I forked dev in the tests initially so i used `customer_testing` instead.

**Comment by GroovinChip** [comment]
Yes, as explained in `CONTRIBUTING.md`, `macos_ui` is developed on Flutter's stable channel. Only the `customer_testing` branch is on `master`. You likely experienced those errors because you are not on Flutter's stable channel.

**Comment by eliasyishak** [comment]
Closing to move to another PR based on `dev`

**Pull request closed without merging**